supervisor/shared/reload.c

```diff
@@ -82,21 +82,21 @@ inline bool autoreload_is_enabled() {
 }
 
 void autoreload_trigger() {
-    if (autoreload_enabled & !autoreload_suspended) {
-        last_autoreload_trigger = supervisor_ticks_ms32();
-        // Guard against the rare time that ticks is 0;
-        if (last_autoreload_trigger == 0) {
-            last_autoreload_trigger += 1;
-        }
-        // Initiate a reload of the VM immediately. Later code will pause to
-        // wait for the autoreload to become ready. Doing the VM exit
-        // immediately is clearer for the user.
+    if (!autoreload_enabled || autoreload_suspended) {
+        return;
+    }
+    bool reload_initiated = autoreload_pending();
+    last_autoreload_trigger = supervisor_ticks_ms32();
+    // Initiate a reload of the VM immediately. Later code will pause to
+    // wait for the autoreload to become ready. Doing the VM exit
+    // immediately is clearer for the user.
+    if (!reload_initiated) {
         reload_initiate(RUN_REASON_AUTO_RELOAD);
     }
 }
 
 bool autoreload_ready() {
-    if (last_autoreload_trigger == 0 || autoreload_suspended != 0) {
```

Comment:
Since `autoreload_suspended` is a bitmask, not a bool. Please do comparisons to zero. I can never remember the truthiness of ints. Doing an explicit comparison is clearer.